client/src/templates/Challenges/components/scene/scene.tsx

```diff
@@ -21,11 +21,32 @@ export function Scene({ scene }: { scene: FullScene }): JSX.Element {
     new Audio(`${sounds}/${audio.filename}${audioTimestamp}`)
   );
 
+  const loadImage = (src: string) => {
+    const img = new Image();
+    img.src = src;
+  };
+
   // on mount
   useEffect(() => {
     const { current } = audioRef;
     current.addEventListener('canplaythrough', audioLoaded);
 
+    // preload images
+    loadImage(`${backgrounds}/${setup.background}`);
+
+    setup.characters.forEach(character => {
+      const characterImages = characterAssets[character.character];
+      Object.values(characterImages).forEach(src => {
+        if (src) {
+          loadImage(src);
+        }
+      });
+    });
+
+    commands.forEach(command => {
+      if (command.background) loadImage(`${backgrounds}/${command.background}`);
+    });
```

Comment:
```suggestion
    commands
      .map(({ background }) =>
        background ? `${backgrounds}/${background}` : null
      )
      .forEach(loadImage);
```
Again, this simplifies the logical flow.